*** checks.c.orig	2010-09-21 10:05:31.000000000 -0500
--- checks.c	2011-01-20 20:13:38.000000000 -0600
***************
*** 3408,3414 ****
  
  #ifdef USE_EVENT_BROKER
  	/* send data to event broker */
! 	broker_host_check(NEBTYPE_HOSTCHECK_PROCESSED,NEBFLAG_NONE,NEBATTR_NONE,temp_host,HOST_CHECK_ACTIVE,temp_host->current_state,temp_host->state_type,start_time_hires,end_time_hires,temp_host->host_check_command,temp_host->latency,temp_host->execution_time,host_check_timeout,queued_check_result->early_timeout,queued_check_result->return_code,NULL,temp_host->plugin_output,temp_host->long_plugin_output,temp_host->perf_data,NULL);
  #endif
  
  	return OK;
--- 3408,3414 ----
  
  #ifdef USE_EVENT_BROKER
  	/* send data to event broker */
! 	broker_host_check(NEBTYPE_HOSTCHECK_PROCESSED,NEBFLAG_NONE,NEBATTR_NONE,temp_host,temp_host->check_type,temp_host->current_state,temp_host->state_type,start_time_hires,end_time_hires,temp_host->host_check_command,temp_host->latency,temp_host->execution_time,host_check_timeout,queued_check_result->early_timeout,queued_check_result->return_code,NULL,temp_host->plugin_output,temp_host->long_plugin_output,temp_host->perf_data,NULL);
  #endif
  
  	return OK;
